src/launchpad/analyzers/android.py

```diff
@@ -36,16 +47,99 @@ def analyze(self, artifact: AndroidArtifact) -> AndroidAnalysisResults:
 
         logger.debug("Found %d APKs", len(apks))
 
-        # TODO: Implement treemap generation from APKs
+        file_analysis = self._get_file_analysis(apks)
+        treemap_builder = TreemapBuilder(
+            app_name=app_info.name,
+            platform="android",
+            # TODO: (Ryan) This is a placeholder, we need to get the actual download compression ratio
+            download_compression_ratio=1.0,
+        )
+
+        treemap = treemap_builder.build_file_treemap(file_analysis)
 
+        analysis_duration = time.time() - start_time
         return AndroidAnalysisResults(
+            generated_at=datetime.now(timezone.utc),
+            analysis_duration=analysis_duration,
             app_info=app_info,
-            file_analysis=FileAnalysis(
-                total_size=0,
-                file_count=0,
-                files_by_type={},
-                duplicate_files=[],
-                largest_files=[],
-            ),
-            analysis_duration=0,
+            treemap=treemap,
+            file_analysis=file_analysis,
+        )
+
+    def _get_file_analysis(self, apks: list[APK]) -> FileAnalysis:
+        file_infos: list[FileInfo] = []
+        total_size = 0
+        path_to_file_info: dict[str, FileInfo] = {}
+
+        for apk in apks:
+            extract_path = apk.get_extract_path()
+            for file_path in extract_path.rglob("*"):
+                if file_path.is_file():
+                    logger.debug("Processing file: %s", file_path)
+                    # Get file extension or use 'unknown' if none
+                    file_type = file_path.suffix.lstrip(".").lower() or "unknown"
+
+                    # Some files have special overrides for the treemap type
+                    if file_path.name in FILE_NAME_TO_TREEMAP_TYPE:
+                        treemap_type = FILE_NAME_TO_TREEMAP_TYPE[file_path.name]
+                    else:
+                        treemap_type = FILE_TYPE_TO_TREEMAP_TYPE.get(file_type, TreemapType.OTHER)
+
+                    # Get relative path from extract directory
+                    relative_path = str(file_path.relative_to(extract_path))
+                    file_size = file_path.stat().st_size
+                    total_size += file_size
+
+                    # If we've seen this path before, merge the sizes to simplify the treemap
+                    # This is intentional as things like the AndroidManifest.xml are duplicated
+                    # across APKs, but to users that's not relevant so we'll group.
+                    if relative_path in path_to_file_info:
```

Comment:
@trevor-e, decided to merge duplicates like `resources.arsc` and `AndroidManifest.xml`. We can always change this in the future. I think this will be the most user-friendly way to view the Treemap